src/pybind/mgr/dashboard/frontend/src/app/shared/directives/form-loading.directive.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  ComponentFactoryResolver,
+  Directive,
+  Input,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+
+import { I18n } from '@ngx-translate/i18n-polyfill';
+
+import { AlertPanelComponent } from '../components/alert-panel/alert-panel.component';
+import { LoadingPanelComponent } from '../components/loading-panel/loading-panel.component';
+import { LoadingStatus } from '../forms/cd-form';
+
+@Directive({
+  selector: '[cdFormLoading]'
+})
+export class FormLoadingDirective {
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private i18n: I18n
+  ) {}
+
+  @Input('cdFormLoading') set cdFormLoading(condition: LoadingStatus) {
+    let factory: any;
+    let content: any;
+
+    this.viewContainer.clear();
+
+    switch (condition) {
```

Comment:
It would be nice if you could test this. It should be testable if you make a test similar to the critical modal component spec, where you create a mock component to explain the usage by example. This will be really helpful if this directive will be expanded in future (passing descriptions and such).

Reply:
done